addons/hr_homeworking/models/hr_homework.py

```diff
@@ -0,0 +1,190 @@
+                            new_days = set(range(start_day,stop_day +1))
+                        days_to_remove = old_days - new_days
+                        days_to_add = new_days - old_days
+                        self._change_days_in_employee_homework(days_to_add, employee_id, work_location_id)
+                        self._change_days_in_employee_homework(days_to_remove, employee_id, False)
+                else:
+                    self.meeting_id.write({
+                        'start_date': vals.get('start_date') or self.start_date,
+                        'stop_date': vals.get('stop_date') or self.stop_date,
+                        'recurrence_update': 'all_events',
+                        })
+                    if self.weekly:
+                        employee_id = self.env['hr.employee'].search([('user_id.id','=',self.env.user.id)])
+                        start_day = self.meeting_id.start_date.weekday()
+                        stop_day = self.meeting_id.stop_date.weekday()
+                        self_start_day = self.start_date.weekday()
+                        self_stop_day = self.stop_date.weekday()
+                        if self_start_day > self_stop_day:
+                            old_days = set(range(self_start_day,7))
+                            old_days.update(set(range(0,self_stop_day +1)))
+                        else:
+                            old_days = set(range(self_start_day,self_stop_day +1))
+                        if start_day > stop_day:
+                            new_days = set(range(start_day,7))
+                            new_days.update(set(range(0,stop_day +1)))
+                        else:
+                            new_days = set(range(start_day,stop_day +1))
+                        days_to_remove = old_days - new_days
+                        days_to_add = new_days - old_days
+                        self._change_days_in_employee_homework(days_to_add, employee_id, self.work_location_id)
+                        self._change_days_in_employee_homework(days_to_remove, employee_id, False)
+            else:
+                if vals.get('work_location_id'):
+                    work_location_id = self.env["hr.work.location"].browse(vals['work_location_id'])
+                    self.meeting_id.write({
+                        'name': work_location_id.icon + ' - '+ work_location_id.name,
+                        'recurrence_update': 'all_events',
+                        })
+                    if self.weekly:
+                        employee_id = self.env['hr.employee'].search([('user_id.id','=',self.env.user.id)])
+                        start_day = self.meeting_id.start_date.weekday()
+                        stop_day = self.meeting_id.stop_date.weekday()
+                        self._write_in_employee_homework(start_day, stop_day, employee_id, work_location_id)
+        res = super().write(vals)
+        return res
+
+    def action_register_day_work_location(self):
```

Comment:
also, if the method is not in batch, good practice to put self.ensure_one(). Also that's being said, I don't know, maybe it makes sense to be in batch :thinking:

Reply:
hum, now it's hard to see, what is the difference (in vals) between the different meetings that you are setting. Maybe it will be more readable if you define generic vals from the start, the vals that are common for all of them, and update it in each case. 

For example 
```
vals = {
    'name': 'Cool Name',
    'start_date': self.start_date,
    # other common values
}

if not self.weekly:
    vals.append({
    # relevant vals
})
else:
    vals.append({
        'end_type': 'forever',
        'recurrency': True,
       # other relevant vals
    })

# some other code
self.env.['calendar.event'].create(vals)
```
This way, when I look at it, I directly see, oh, so it's it's not weekly the difference is that we set reccurence to true, or end_type.

Just and suggestion :smile: